subgraph/src/public-lock.ts

```diff
@@ -127,12 +128,16 @@ export function handleTransfer(event: TransferEvent): void {
     const key = Key.load(keyID)
     if (key) {
       key.owner = event.params.to
-      key.expiration = getKeyExpirationTimestampFor(
+      const expiration = getKeyExpirationTimestampFor(
         event.address,
         event.params.tokenId,
         event.params.to
       )
 
+      key.expiration = expiration
+      // Mark as cancelled if the key has expired
+      key.cancelled = expiration.lt(event.block.timestamp)
```

Comment:
I think we should removed the `cancelled` property as it creates a lot of confusion.

Reply:
They are different. Cancelled is only true if the key was cancelled or transferred while expiration always follows the key expiration.